plugins/azure/frontdoor/frontDoorWafDefaultRuleSet.js

```diff
@@ -0,0 +1,69 @@
+    recommended_action: 'Modify the Front Door WAF policy and add latest default rule set.',
+    link: 'https://learn.microsoft.com/en-us/azure/web-application-firewall/afds/afds-overview#waf-policy-and-rules',
+    apis: ['afdWafPolicies:listAll'],
+    realtime_triggers: ['microsoftnetwork:frontdoorwebapplicationfirewallpolicies:write', 'microsoftnetwork:frontdoorwebapplicationfirewallpolicies:delete'],
+
+    run: function(cache, settings, callback) {
+        const results = [];
+        const source = {};
+        const locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.afdWafPolicies, (location, rcb) => {
+
+            var afdWafPolicies = helpers.addSource(cache, source,
+                ['afdWafPolicies', 'listAll', location]);
+
+            if (!afdWafPolicies) return rcb();
+
+            if (afdWafPolicies.err || !afdWafPolicies.data) {
+                helpers.addResult(results, 3, 'Unable to query for Front Door WAF policies: ' + helpers.addError(afdWafPolicies), location);
+                return rcb();
+            }
+            if (!afdWafPolicies.data.length) {
+                helpers.addResult(results, 0, 'No existing Front Door WAF policies found', location);
+                return rcb();
+            }
+
+            var premiumWAF = false;
+
+            for (let policy of afdWafPolicies.data) {
+                if (!policy.id || (policy.sku && policy.sku.name != 'Premium_AzureFrontDoor')) continue;
+
+                premiumWAF = true;
+                var ruleSet = (policy.managedRules &&
+                    policy.managedRules.managedRuleSets &&
+                    policy.managedRules.managedRuleSets.find(set => set.ruleSetType && set.ruleSetType.toLowerCase() === 'microsoft_defaultruleset')) || {};
+
+                var ruleSetVersion = ruleSet.ruleSetVersion ? parseFloat(ruleSet.ruleSetVersion) : '';
+                var ruleSetAction = ruleSet.ruleSetAction ? ruleSet.ruleSetAction.toLowerCase() : '';
+
+                if (ruleSetVersion >= 2.1) {
+                    if (ruleSetAction == 'block') {
+                        helpers.addResult(results, 0, `Front Door WAF policy has latest ${ruleSet.ruleSetType}: ${ruleSet.ruleSetVersion} default rule set configured with ${ruleSetAction} action`, location, policy.id);
+                    } else {
+                        helpers.addResult(results, 2, `Front Door WAF policy has latest ${ruleSet.ruleSetType}: ${ruleSet.ruleSetVersion} default rule set configured with ${ruleSetAction} action`, location, policy.id);
+                    }
+                } else {
+                    helpers.addResult(results, 2, 'Front Door WAF policy does not have latest default rule set configured', location, policy.id);
+                }
+            }
+            
+            if (!premiumWAF) {
+                helpers.addResult(results, 0, 'No existing Front Door WAF policies found', location);
```

Comment:
```suggestion
                helpers.addResult(results, 0, 'No existing Front Door WAF policies found', location);
```
this message should either say that no premium policies found or we should produce a pass result for non-premium policies